### PR TITLE
Add CI workflow on GitHub Actions

### PR DESCRIPTION
**Pull request by junghans**
Once merged you will see CI like this: https://github.com/junghans/kokkos-fortran-interop/actions/runs/299263934

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        gcc_ver: ['8', '9']
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install deps
+        run: sudo apt-get install libhwloc-dev
+      - name: Checkout kokkos
+        uses: actions/checkout@v2.2.0
+        with:
+          repository: kokkos/kokkos
+          ref: 3.0.00
```
Comment (dalg24) [inline .github/workflows/CI.yml:29]: Shouldn't  this be `matrix.kokkos_ver`?

Comment (junghans) [reply .github/workflows/CI.yml:29]: Good catch.

**Comment by junghans** [comment]
https://github.com/junghans/kokkos-fortran-interop/actions/runs/303105072

**Comment by womeld** [comment]
Does github have any facility to check against CUDA builds of Kokkos? Or use the Intel or XL compilers?

I am grateful for this! Most certainly, just curious if there are more capabilities and whether I could ditch the homegrown "CI" I use locally.

**Comment by junghans** [comment]
XL no, but there is some hackery to make intel work, but it will need a very different setup, hence let’s do this in a separate PR.

**Review by dalg24 — APPROVED** [review]
I am not familiar with GH Actions but the make looks fine to me

One minor comment:  you enable all warnings but do not pass `-Werror` nor parse the log to ensure no warnings are emitted

**Comment by junghans** [comment]
Yeah, I wanted to leave @womeld some time to fix the warnings first ;-)

**Comment by junghans** [comment]
@womeld ping, anything I need to do here?

**Comment by womeld** [comment]
My only complaint was that github doesn't do exotic stuff, I think. So I don't see a reason not to merge it. You are most certainly the CI expert, not me! This might catch simple things, and won't hurt, so a net positive.

**Comment by womeld** [comment]
Thank you for the reminder, and the effort !

**Comment by junghans** [comment]
Yeah, https://github.com/kokkos/kokkos-fortran-interop/actions/runs/336592924

**Pull request merged**